src/Compiler.Dynamic/SystemWebCompilation.cs

```diff
@@ -165,20 +165,12 @@ private async Task<T[]> GetDependencyPages(TemplateParser parser, CancellationTo
     {
         var p = new List<Task<T>>();
 
-        foreach (var path in GetDependencyPaths(parser))
+        foreach (var path in parser.GetDependencyPaths())
```

Comment:
will this get the HeaderFile as dependency ?

Reply:
what's "HeaderFile"?

Reply:
UserControl.. sorry sometime mix with other programming languages.

Reply:
no - I didn't add any explicit support for usercontrol stuff, but should be a lot easier now